arm/Microsoft.Authorization/policyDefinitions/.bicep/nested_policyDefinitions_mg.bicep

```diff
@@ -1,30 +1,57 @@
 targetScope = 'managementGroup'
-param policyDefinitionName string
+
+@sys.description('Required. Specifies the name of the policy definition. Space characters will be replaced by (-) and converted to lowercase')
+@maxLength(64)
+param name string
+
+@sys.description('Optional. The display name of the policy definition.')
 param displayName string = ''
-param policyDescription string = ''
+
+@sys.description('Optional. The policy definition description.')
+param description string = ''
+
+@sys.description('Optional. The policy definition mode. Default is All, Some examples are All, Indexed, Microsoft.KeyVault.Data.')
+@allowed([
+  'All'
+  'Indexed'
+  'Microsoft.KeyVault.Data'
+  'Microsoft.ContainerService.Data'
+  'Microsoft.Kubernetes.Data'
+])
 param mode string = 'All'
+
+@sys.description('Optional. The policy Definition metadata. Metadata is an open ended object and is typically a collection of key-value pairs.')
 param metadata object = {}
+
+@sys.description('Optional. The policy definition parameters that can be used in policy definition references.')
 param parameters object = {}
+
+@sys.description('Required. The Policy Rule details for the Policy Definition')
 param policyRule object
+
+@sys.description('Required. The group ID of the Management Group')
 param managementGroupId string
-param location string = deployment().location
 
-var policyDefinitionName_var = toLower(replace(policyDefinitionName, ' ', '-'))
+var name_var = toLower(replace(name, ' ', '-'))
 
-resource policyDefinition 'Microsoft.Authorization/policyDefinitions@2020-09-01' = {
-  name: policyDefinitionName_var
-  location: location
+resource policyDefinition 'Microsoft.Authorization/policyDefinitions@2021-06-01' = {
+  name: name_var
   properties: {
     policyType: 'Custom'
     mode: mode
-    displayName: (empty(displayName) ? null : displayName)
-    description: (empty(policyDescription) ? null : policyDescription)
-    metadata: (empty(metadata) ? null : metadata)
-    parameters: (empty(parameters) ? null : parameters)
+    displayName: empty(displayName) ? null : displayName
```

Comment:
Can you flip these to e.g. `!empty(displayName) ? displayName : null` ? Makes it more readable to have the default in the back

Reply:
Same question for the other templates

Reply:
addressed on all other templates and PRs